dbms/src/Common/BackgroundTask.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/BackgroundTask.h>
+
+#include <fstream>
+namespace DB
+{
+bool process_mem_usage(double & resident_set)
```

Comment:
```suggestion
std::optional<double> getProcessMemUsage()
```

use `std::optional` is safer and easier to use.